gazebo_ros/src/gazebo_ros_init.cpp

```diff
@@ -188,25 +195,29 @@ void GazeboRosInit::Load(int argc, char ** argv)
 void GazeboRosInitPrivate::onPerformanceMetrics(
   ConstPerformanceMetricsPtr & msg)
 {
-  gazebo_msgs::msg::PerformanceMetrics msg_ros;
-  msg_ros.header.stamp = Convert<builtin_interfaces::msg::Time>(world_->SimTime());
-  msg_ros.real_time_factor = msg->real_time_factor();
-  for (auto sensor : msg->sensor()) {
-    gazebo_msgs::msg::SensorPerformanceMetric sensor_msgs;
-    sensor_msgs.sim_update_rate = sensor.sim_update_rate();
-    sensor_msgs.real_update_rate = sensor.real_update_rate();
-    sensor_msgs.name = sensor.name();
-
-    if (sensor.has_fps()) {
-      sensor_msgs.fps = sensor.fps();
-    } else {
-      sensor_msgs.fps = -1;
+  // Check if performance metrics parameter was enabled
+  bool check_enable_performance_metrics;
+  this->ros_node_->get_parameter("enable_performance_metrics", check_enable_performance_metrics);
+  if (check_enable_performance_metrics) {
```

Comment:
Prefer returning early, this cuts down on the level of indentation (and is more readable IMO). E.g.

```c++
if (!check_enable_performance_metrics) {
  return;
}

Reply:
Done